docs/end_to_end_tutorials/dev_practices/e2e_evaluation.md

```diff
@@ -0,0 +1,52 @@
+# End-to-End Evaluation
+End-to-End evaluation should be the guiding signal for your RAG application - will my pipeline generate the right responses given the data sources and a set of queries?
+
+While it helps initially to individually inspect queries and responses, as you deal with more failure and corner cases, it may stop being feasible to look at each query individually, and rather it may help instead to define a set of summary metrics or automated evaluation, and gain an intuition for what they might be telling you and where you might dive deeper.
+
+## Setting up an Evaluation Set
+
+It is helpful to start off with a small but diverse set of queries, and build up more examples as one discovers problematic queries or interactions.
+
+We've created some tools that automatically generate a dataset for you given a set of documents to query. (See example below).
+
+
+```{toctree}
+---
+maxdepth: 1
+---
+/examples/evaluation/QuestionGeneration.ipynb
+```
+
+In the future, we will also be able to create datasets automatically against tools.
+
+## Qualitative v.s. Quantitative Eval
```

Comment:
nit: this section feels a bit weird because 1) quantitative and qualititave eval as concepts aren't defined, and 2) there's no action item to click into one of our notebooks (do we show any of these comparisons?)

Reply:
Right, I think it's actually more "metrics based v.s. bert based v.s. LLM-based v.s. Human annotator based". I guess we can be more specific about that and present the options as a spectrum (it's not really an if/else)